input/postgres/statements.go

```diff
@@ -83,25 +83,56 @@ func GetStatements(server *state.Server, logger *util.Logger, db *sql.DB, global
 	var totalTimeField string
 	var optionalFields string
 	var sourceTable string
+	var extSchema string
+	var extVersion float32 // defined in pg_stat_statements.c
+	var foundExtVersion float32
 
 	if postgresVersion.Numeric >= state.PostgresVersion13 {
-		totalTimeField = statementSQLpg13TotalTimeField
+		extVersion = 1.8
+	} else if postgresVersion.Numeric >= state.PostgresVersion95 {
+		extVersion = 1.3
+	} else if postgresVersion.Numeric >= state.PostgresVersion94 {
+		extVersion = 1.2
 	} else {
-		totalTimeField = statementSQLDefaultTotalTimeField
+		extVersion = 1.1
 	}
 
-	if postgresVersion.Numeric >= state.PostgresVersion13 {
+	err = db.QueryRow(QueryMarkerSQL+"SELECT nspname, extversion FROM pg_extension pge INNER JOIN pg_namespace pgn ON pge.extnamespace = pgn.oid WHERE pge.extname = 'pg_stat_statements'").Scan(&extSchema, &foundExtVersion)
+	if err != nil && err != sql.ErrNoRows {
+		return nil, nil, nil, err
+	}
+
+	if err == sql.ErrNoRows {
+		logger.PrintInfo("pg_stat_statements does not exist, trying to create extension...")
+		_, err = db.Exec(QueryMarkerSQL + "CREATE EXTENSION IF NOT EXISTS pg_stat_statements SCHEMA public")
```

Comment:
This is outside of the scope of the PR, but as a side note, we should stop doing this. It rarely works (since the monitoring user typically doesn't have permissions to create extensions), and when it does, do we really want this behavior as a side effect of running the collector?

Reply:
Okay, well it's easy to remove. @lfittl thoughts?

Reply:
Will go ahead and merge this as-is. We can follow up with that change in a future PR

Reply:
Yeah, lets discuss separately - I remember there being cases where this matters (maybe it was Heroku?) -- essentially when there is no way to create a separate monitoring user, this ensures that you don't have to run an extra step during setup (since you'll have to run the collector with database owner permissions anyway, due to platform limitations)